src/reql/executor/abstract_value.cr

```diff
@@ -213,6 +213,19 @@ module ReQL
       end
     end
 
+    def float64_value
+      case v = number_value
+      when Int32
+        v.to_f64
+      when Int64
+        v.to_f64
+      when Float64
+        v
+      else
+        raise ReQL::QueryLogicError.new "Number not an Float: #{v}"
+      end
+    end
```

Comment:
´number_value` returns a `Int32 | Int64 | Float64` and all of them respond to `.to_f64`. This method can be simplified to

```cr
def float64_value
  number_value.to_f64
end
```